### PR TITLE
[Log] enhances security logs

### PR DESCRIPTION
**Pull request by Zowac**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Fixed issues | comma-separated list of issues fixed by the PR, if any

Feeback from #1537

### CODE REVIEW

File: src/main/app/Resources/translations/security.fr.json [hunk 1]
```diff
@@ -11,5 +11,18 @@
   "newPassword": "L'utilisateur username a modifié son mot de passe.",
   "viewAs": "L'utilisateur username a utilisé la fonctionnalité 'voir en tant que' avec le Role role",
   "validateEmail": "L'utilisateur username a validé son adresse email",
-  "authenticationFailure": "Connexion échoué pour l'utilisateur username : message"
+  "authenticationFailure": "Connexion échoué pour l'utilisateur username : message",
+  "switchUser": "L'utilisateur username a utilisé la fonctionnalité 'voir en tant que' avec l'utilisateur target",
+  "event.security.add_role": "evenement.securite.ajout_role",
```
Comment (Elorfin) [inline src/main/app/Resources/translations/security.fr.json:16]: Should be human readable : "Ajout d'un rôle", "Connexion utilisateur" etc.
You will also need the english translations.

File: src/main/core/Subscriber/SecurityEventSubscriber.php [hunk 2]
```diff
@@ -48,18 +50,40 @@ public static function getSubscribedEvents(): array
...
+            SecurityEvents::SWITCH_USER => 'logEventSwitchUser',
         ];
     }
 
     public function logEvent(Event $event, string $eventName): void
     {
         $logEntry = new SecurityLog();
         $logEntry->setDetails($event->getMessage($this->translator));
-        $logEntry->setEvent($eventName);
+        $logEntry->setEvent($this->translator->trans($eventName, [], 'security'));
```
Comment (Elorfin) [inline src/main/core/Subscriber/SecurityEventSubscriber.php:61]: You should store the real event and do the translation in the ui.

File: src/main/core/Subscriber/SecurityEventSubscriber.php [hunk 3]
```diff
@@ -48,18 +50,40 @@ public static function getSubscribedEvents(): array
...
+            $logEntry = new SecurityLog();
+            $logEntry->setDetails($this->translator->trans(
+                'switchUser',
+                [
+                    'username' => $this->security->getUser(),
+                    'target' => $event->getTargetUser(),
+                ],
+                'security'
+            ));
+            $logEntry->setEvent($this->translator->trans($eventName, [], 'security'));
```
Comment (Elorfin) [inline src/main/core/Subscriber/SecurityEventSubscriber.php:81]: You should store the real event and do the translation in the ui.

**Review by Elorfin — CHANGES_REQUESTED** [review]

File: src/main/core/Subscriber/SecurityEventSubscriber.php (hunk 2 above)
Comment (Elorfin) [reply src/main/core/Subscriber/SecurityEventSubscriber.php:61]: An example of translated col in ListData : https://github.com/claroline/Claroline/blob/13.x/src/main/core/Resources/modules/modals/template-types/components/modal.jsx#L34

File: src/main/app/Resources/translations/security.fr.json (hunk 1 above)
Comment (Zowac) [reply src/main/app/Resources/translations/security.fr.json]: Change done, thx :)

File: src/main/core/Resources/translations/platform.en.json [hunk 4]
```diff
@@ -337,6 +337,18 @@
...
+    "event.security.user_login": "User login",
+    "event.security.user_logout": "User logout",
+    "event.security.user_enable": "User enable",
+    "event.security.user_disable": "User disable",
+    "event.security.forgot_password": "Forgot password",
+    "event.security.new_password": "New password",
+    "event.security.view_as": "View as",
+    "security.switch_user": "View as",
+    "event.security.validate_email": "Validate email",
+    "event.security.authentication_failure": "Authentication failure",
```
Comment (Elorfin) [inline src/main/core/Resources/translations/platform.en.json:351]: It was better in `security` domain.

File: src/main/core/Resources/translations/platform.fr.json [hunk 5]
```diff
@@ -342,6 +342,18 @@
...
+    "event.security.user_login": "Connexion utilisateur",
+    "event.security.user_logout": "Deconnexion utilisateur",
+    "event.security.user_enable": "Activation utilisateur",
+    "event.security.user_disable": "Désactivation utilisateur",
+    "event.security.forgot_password": "Mot de passe oublié",
+    "event.security.new_password": "Nouveau mot de passe",
+    "event.security.view_as": "Voir en tant que",
+    "security.switch_user": "Voir en tant que",
+    "event.security.validate_email": "Validation de l'email",
+    "event.security.authentication_failure": "Authentification échouée",
```
Comment (Elorfin) [inline src/main/core/Resources/translations/platform.fr.json:356]: It was better in `security` domain.

File: src/plugin/analytics/Resources/modules/administration/dashboard/components/log.jsx [hunk 6]
```diff
@@ -30,20 +35,15 @@ const DashboardLog = () =>
...
           label: trans('target'),
-          displayed: true
+          displayed: false
         }, {
           name: 'event',
-          type: 'string',
+          type: 'translation',
           label: trans('event'),
-          displayed: true
+          displayed: false
```
Comment (Elorfin) [inline src/plugin/analytics/Resources/modules/administration/dashboard/components/log.jsx:46]: `options: {domain: 'security'}` to follow changes requested in translations.

**Review by Elorfin — CHANGES_REQUESTED** [review]

File: src/plugin/analytics/Resources/modules/administration/dashboard/components/log.jsx (hunk 6 above)
Comment (Zowac) [reply src/plugin/analytics/Resources/modules/administration/dashboard/components/log.jsx]: Done, thx :)

**Review by Elorfin — APPROVED** [review]

**Pull request merged**